### PR TITLE
[DOCS] Sync missing definitions from Elasticsearch glossary

### PR DESCRIPTION
**Pull request by jrodewig**
Adds a few [Elasticsearch definitions](https://www.elastic.co/guide/en/elasticsearch/reference/current/glossary.html) missing from the [Stack Docs glossary](https://www.elastic.co/guide/en/elastic-stack-glossary/current/terms.html):

- [Cross-cluster search](https://github.com/elastic/elasticsearch/blame/master/docs/reference/glossary.asciidoc#L36)
- [Filter](https://github.com/elastic/elasticsearch/blame/master/docs/reference/glossary.asciidoc#L69)
- [Query](https://github.com/elastic/elasticsearch/blame/master/docs/reference/glossary.asciidoc#L140)

Uncovered this while working on these issues:
- elastic/elasticsearch#40560
- elastic/docs#759

### CODE REVIEW

File: docs/en/glossary/glossary.asciidoc [hunk 1]
```diff
@@ -185,6 +185,14 @@ local cluster. For more information, see {stack-ov}/xpack-ccr.html[{ccr-cap}].
 +
 //Source: X-Pack
 endif::xpack-terms[]
+ifdef::elasticsearch-terms[]
+[[glossary-ccs]] {ccs} (CCS)::
+
+The {ccs} feature enables any node to act as a federated client across
+multiple clusters. See {ref}/modules-cross-cluster-search.html[Cross-cluster search].
```
Comment (lcawl) [inline docs/en/glossary/glossary.asciidoc]: The hyphenation has changed recently, so I recommend using attributes for all CCS occurrences:
```suggestion
multiple clusters. See {ref}/modules-cross-cluster-search.html[{ccs-cap}].   
```

Comment (jrodewig) [reply docs/en/glossary/glossary.asciidoc]: Good catch! Pushed e948189 to update.

File: docs/en/glossary/glossary.asciidoc [hunk 2]
```diff
@@ -550,6 +572,21 @@ nodes handling the user requests.
 //Source: Cloud
 endif::cloud-terms[]
 ifdef::elasticsearch-terms[]
+[[glossary-query]] query ::
+
+A query is the basic component of a search. A search can be defined by one or
```
Comment (lcawl) [inline docs/en/glossary/glossary.asciidoc]: These feel more verbose than I'd expect for a glossary entry, but I guess it doesn't matter too much.

Comment (jrodewig) [reply docs/en/glossary/glossary.asciidoc]: Agree. I think a rewrite is probably in order. Query isn't actually defined here IMO.

I opened elastic/elasticsearch#40643 to address that separately. I'm planning on doing a general query overhaul shortly. That issue will likely be a part of that work effort.

**Pull request merged**